crates/construction/src/manufacturing.rs

```diff
@@ -335,6 +342,16 @@ fn change_locations(
                 event.factory(),
                 event.position(),
             ));
+            let Ok((transform, object_type)) = transforms.get(event.factory()) else {
+                continue;
+            };
+            let solid = solids.get(*object_type);
+            let Some(factory) = solid.factory() else {
```

Comment:
I would unwrap here since the component `DeliveryLocation` implies that it is a factory.